teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md

```diff
@@ -171,12 +185,277 @@ In addition to grouping related resources, Tekton Results provides long term sto
 of `PipelineRuns`, `TaskRuns` and `Runs` away from the runtime storage in etcd.
 Read more in [TEP-0021][tep-0021].
 
-## Open Questions
+## Proposal
 
-* [Qn #1][qn-1]: What fields will no longer be available? What do we lose from the optimizations?
-  We will discuss this in the proposal.
-* [Qn #2][qn-1]: What does an example status look like?
-  We will provide examples in the proposal.
+The `PipelineRunStatus` should contain the names and overall status (`ConditionSucceeded`)
+of its `TaskRuns` and `Runs`. Users and tools can find the complete status of `TaskRuns`
+and `Runs` in the cluster. Moreover, they can rely on [Tekton Results](#tekton-results-api)
+and [Owner References and Labels](#owner-references-and-labels) to identify related objects.
+
+### API Changes
+
+1. Rename the existing structs that embed the complete status of `TaskRuns` and `Runs`:
+   * `PipelineRunTaskRunStatus` to `PipelineRunTaskRunEmbeddedStatus`
+   * `PipelineRunRunStatus` to `PipelineRunRunEmbeddedStatus`
+
+2. Introduce two new structs to store the minimal status of `TaskRuns` and `Runs` in the
+`PipelineRun` status, with the names and overall status only:
+
+```go
+type PipelineRunTaskRunStatus struct {
+	TaskRunName string `json:"taskRunName,omitempty"`
+	duckv1beta1.Status `json:",inline"`
+}
+
+type PipelineRunRunStatus struct {
+	RunName string `json:"runName,omitempty"`
+	duckv1beta1.Status `json:",inline"`
+}
+```
```

Comment:
I like this approach is extensible to support `PipelineRuns` and other types we may support in `PipelineTasks` in addition to `TaskRuns` and `Runs`. However, it's more verbose because the type and kind have to be repeated for each `TaskRun` and `Run`. It also makes it more difficult for tools and users to parse for specific type of resource when looking at the status (compared to separate fields).

Added it as an alternative in the TEP - happy to discuss this further :)

Reply:
I'm not personally concerned with verbosity here since these will be server generated, and leaning on TypeMeta may make this type easier to use with client-go libraries since we'll use the existing well-known TypeMeta/GVK types.
This also more in line with https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#multiple-resource-reference

Not sure if this is intentional behavior, but what got me thinking about is this is a PipelineRun like this:

```yaml
apiVersion: tekton.dev/v1beta1
kind: PipelineRun
metadata:
  generateName: pr-
spec:
  pipelineSpec:
    tasks:
      - name: a
        taskSpec:
          apiVersion: tekton.dev/v1alpha1
          steps:
            - image: alpine
              script: |
                echo "hi!"
      - name: b
        taskSpec:
          steps:
            - image: alpine
              script: |
                echo "hello!"
```

Now today we ignore the TypeMeta information, but it is part of our API. 
Debatable whether this is expected behavior or a bug - I can see it being a valid use case that I want to use alpha tasks to take advantage of some experimental behavior within a v1beta1 Pipeline. Maybe I want to use v2alpha1, etc. How we want to resolve this is a separate discussion - I can open an issue for this.